File: 1328-break-a-palindrome/1328-break-a-palindrome.py
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        pal, n, ans = palindrome, len(palindrome), ""
        
        if n == 1: return "" 
        for i in range(n//2):
            if pal[i] != 'a':
                return pal[:i] + 'a' + pal[i+1:]
            
        return pal[:-1] + 'b'

File: 1328-break-a-palindrome/1328-break-a-palindrome.cpp
class Solution {
public:
    bool ispal(string s) {
        string t = s;
        reverse(t.begin(), t.end());
        return s == t;
    }
    string breakPalindrome(string palindrome) {
        if (palindrome.size() == 1)
            return "";
        bool ok = true;
        for (auto &n : palindrome) {
            if (n > 'a') {
                char c = n;
                n = 'a';
                if (!ispal(palindrome)) {
                    ok = false;
                    break;
                }
                n = c;
            }
        }
        if (ok)
            palindrome.back() = 'b';
        return palindrome;
    }
};